rsocket-examples/src/main/java/io/rsocket/examples/transport/ws/WebSocketRouteTransportHeadersSample.java

```diff
@@ -0,0 +1,138 @@
+                    .filteringInbound(
+                        headers -> headers.containsValue("Authorization", "test", true))
+                    .closingWithStatus(headers -> new WebSocketCloseStatus(4404, "Unauthorized"))
+                    .observingOn("/")
+                    .build(HttpServer.create().host("localhost").port(8080)))
+            .start()
+            .block();
+
+    WebsocketClientTransport clientTransport =
+        WebsocketClientTransport.create(disposableServer.address());
+
+    MonoProcessor<WebSocketCloseStatus> statusMonoProcessor = MonoProcessor.create();
+    clientTransport.setTransportHeaders(
+        () -> {
+          HashMap<String, String> map = new HashMap<>();
+          map.put("Authorization", "1");
+          return map;
+        });
+
+    clientTransport.setCloseStatusConsumer(
+        webSocketCloseStatusMono -> webSocketCloseStatusMono.log().subscribe(statusMonoProcessor));
+
+    RSocket socket =
+        RSocketFactory.connect()
+            .keepAliveAckTimeout(Duration.ofMinutes(10))
+            .frameDecoder(PayloadDecoder.ZERO_COPY)
+            .transport(clientTransport)
+            .start()
+            .block();
+
+    try {
+      Flux.range(0, 100).concatMap(i -> socket.fireAndForget(payload1.retain())).blockLast();
+
+    } catch (Exception e) {
+      System.out.println("Observed WebSocket Close Status " + statusMonoProcessor.peek());
+    }
+
+    socket.dispose();
+
+    WebsocketClientTransport clientTransport2 =
+        WebsocketClientTransport.create(disposableServer.address());
+
+    clientTransport2.setTransportHeaders(
+        () -> {
+          HashMap<String, String> map = new HashMap<>();
+          map.put("Authorization", "test");
+          return map;
+        });
+
+    RSocket rSocket =
+        RSocketFactory.connect()
+            .keepAliveAckTimeout(Duration.ofMinutes(10))
+            .frameDecoder(PayloadDecoder.ZERO_COPY)
+            .transport(clientTransport2)
+            .start()
+            .block();
+
+    // expect normal execution here
+    rSocket.requestResponse(payload1).block();
+  }
+
+  private static class SocketAcceptorImpl implements SocketAcceptor {
+    @Override
+    public Mono<RSocket> accept(ConnectionSetupPayload setupPayload, RSocket reactiveSocket) {
+      return Mono.just(
+          new AbstractRSocket() {
+
+            @Override
+            public Mono<Void> fireAndForget(Payload payload) {
+              //                  System.out.println(payload.getDataUtf8());
+              payload.release();
+              return Mono.empty();
+            }
+
+            @Override
+            public Mono<Payload> requestResponse(Payload payload) {
+              return Mono.just(payload);
+            }
+
+            @Override
+            public Flux<Payload> requestChannel(Publisher<Payload> payloads) {
+              return Flux.from(payloads).subscribeOn(Schedulers.single());
```

Comment:
Is the subscribe on required here?

Reply:
Actually it is not necessarily needed, thus just removed it at all